include/cuda/std/detail/libcxx/include/span

```diff
@@ -528,7 +528,7 @@ auto as_writable_bytes(span<_Tp, _Extent> __s) noexcept
 
 #endif // _LIBCUDACXX_STD_VER > 11
 
-#if _LIBCUDACXX_STD_VER > 14
+#if _LIBCUDACXX_STD_VER > 14 && !defined(_LIBCUDACXX_HAS_NO_DEDUCTION_GUIDES)
```

Comment:
@miscco GCC-6 strikes again. 🙃

Reply:
Yeah, I fixed that locally on my branch. That said, I thought we decided to settle on gcc-7 as the minimal gcc version, so we might as well just drop it completely